### PR TITLE
MM-31913 Fix assignee dropdown

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Shift the assignee dropdown up if it goes past the window's `innerHeight`

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-31913

### CODE REVIEW

**Review by lieut-data — APPROVED** [review]
Magic! Worth a unit test for this edge case? (Unless I missed it...)

File: webapp/src/hooks.ts [hunk 1]
```diff
@@ -95,3 +95,15 @@ export function useTimeout(callback: () => void, delay: number | null) {
     // In case you want to manually clear the timeout from the consuming component...:
     return timeoutRef;
 }
+
+export function useClientRect() {
```
[the remark this answers is not in the document]
Comment (cpoile) [reply webapp/src/hooks.ts:99]: Thought about that, but if you resize the window, you'll have to click outside the profile and it will disappear, so not a problem for this use. And I wasn't sure if we wanted to add complexity for something that /might/ cause a problem sometime in the future.

**Comment by cpoile** [comment]
@lieut-data I'm actually not really sure how we would be able to test this -- I couldn't find a "div is completely visible and none is hidden", and Harrison and Nick both can't think of a test that would show that it's not covered up. Can you think of anything?

**Comment by lieut-data** [comment]
I'm reaching here, since I haven't had to test this exact case myself, but I seem to recall Cypress complaining about things being "invisible" when they past the fold, i.e. why we had to increase the window size on some tests. Does it suffice to generate an incident with a number of tasks (so they fill up the whole vertical space), and then click on the last one's assignee, and then assert that a known user appears in the list (which previously would have rendered below the fold)? 

0/5 on this blocking the review.

**Comment by cpoile** [comment]
@lieut-data Found a way to test if a window is within the viewport. Pretty straightforward. I don't think it's testing if the window is obscured by another div (which is what Harrison and Nick were talking about) but it does what we want.

**Review by lieut-data — APPROVED** [review]

File: webapp/src/components/profile/profile_selector.tsx [hunk 2]
```diff
@@ -148,6 +151,26 @@ export default function ProfileSelector(props: Props) {
...
+
+        const innerHeight = window.innerHeight;
+        const numProfilesShown = Math.min(6, userOptions.length);
+        const spacePerProfile = 48;
+        const dropdownYShift = 27;
+        const dropdownReqSpace = 80;
+        const extraSpace = 10;
+        const dropdownBottom = rect.top + dropdownYShift + dropdownReqSpace + (numProfilesShown * spacePerProfile) + extraSpace;
+        setMoveUp(Math.max(0, dropdownBottom - innerHeight));
+    }, [rect, userOptions]);
```
Comment (crspeller) [inline webapp/src/components/profile/profile_selector.tsx:172]: Should the second one be `userOptions.length` to be more specific?

File: webapp/src/hooks.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (crspeller) [reply webapp/src/hooks.ts:99]: Maybe a comment to note this limitiation?

File: webapp/src/components/profile/profile_selector.tsx [hunk 3]
```diff
@@ -233,26 +268,34 @@ interface DropdownProps {
...
+    const classes = classNames('IncidentFilter', 'profile-dropdown',
+        'IncidentFilter--active', 'profile-dropdown--active', {'show-on-right': showOnRight});
 
-const Menu = (props: Record<string, any>) => {
+    const top = 27 - (moveUp || 0);
     return (
-        <div {...props}/>
+        <div
+            className={classes}
+            css={{position: 'relative'}}
```
Comment (crspeller) [inline webapp/src/components/profile/profile_selector.tsx:288]: This would look much nicer if it was using styled components :)

File: webapp/src/hooks.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cpoile) [reply webapp/src/hooks.ts:99]: Sounds good, done!

File: webapp/src/components/profile/profile_selector.tsx (hunk 3 above)
Comment (cpoile) [reply webapp/src/components/profile/profile_selector.tsx]: Yes, you're right -- fixed for the first one, but not the second (I did it, but there was so much boilerplate to get the props into the styled component that it was not worth it, vs one line of `css=`). Also moved `Blanket` into a styled component.
